src/backend/catalog/storage.c

```diff
@@ -122,24 +127,27 @@ RelationCreateStorage(RelFileNode rnode, char relpersistence, char relstorage)
 /*
  * Perform XLogInsert of a XLOG_SMGR_CREATE record to WAL.
  */
-void
-log_smgrcreate(RelFileNode *rnode, ForkNumber forkNum)
+XLogRecPtr
+log_smgrcreate(RelFileNode *rnode, ForkNumber forkNum, char relstorage)
 {
-	xl_smgr_create xlrec;
+	xl_smgr_create_pdl xlrec;
 	XLogRecData rdata;
 
 	/*
 	 * Make an XLOG entry reporting the file creation.
 	 */
-	xlrec.rnode = *rnode;
+	xlrec.relnode.node = *rnode;
+	/* temp relations are not logged in WAL, so it is always false here */
+	xlrec.relnode.isTempRelation = false;
+	xlrec.relnode.relstorage = relstorage;
 	xlrec.forkNum = forkNum;
 
 	rdata.data = (char *) &xlrec;
 	rdata.len = sizeof(xlrec);
 	rdata.buffer = InvalidBuffer;
 	rdata.next = NULL;
 
-	XLogInsert(RM_SMGR_ID, XLOG_SMGR_CREATE, &rdata);
+	return XLogInsert(RM_SMGR_ID, XLOG_SMGR_CREATE_PDL, &rdata);
```

Comment:
Why not just do this:
```suggestion
	XLogFlush(XLogInsert(RM_SMGR_ID, XLOG_SMGR_CREATE_PDL, &rdata));
```

Reply:
Because it is currently requested by the SRS, that `log_smgrcreate()` returns XLogRecPtr, and flush is done by the caller function.